### PR TITLE
Simplify TokenReview cache

### PR DESCRIPTION
**Pull request by jlpadilla**
Signed-off-by: Jorge Padilla <jpadilla@redhat.com>

**Related Issue:**  stolostron/backlog#22507

### Description of changes
- Simplifies the TokenReview cache implementation.

### CODE REVIEW

File: pkg/rbac/authenticationMiddleware_test.go [hunk 1]
```diff
@@ -10,7 +10,13 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-//test valid token from cookie
+// Runs before the tests
+func TestMain(m *testing.M) {
+	// Replace the cache with a mock cache with a fake kubernetes client.
+	cache = newMockCache()
+}
```
Comment (jlpadilla) [inline pkg/rbac/authenticationMiddleware_test.go:17]: Test were using a real connection to the kubernetes cluster, which created inconsistent results.  This replaces the real connection with a mock client.

@Anxhela21 FYI, this resolves the problem of tests failing when running in prow.

File: pkg/rbac/authenticationMiddleware_test.go [hunk 2]
```diff
@@ -44,8 +50,8 @@ func TestTokenInvalidCookieAuthenticated(t *testing.T) {
 	authen := AuthenticateUser(authenticateHandler)
 
 	authen.ServeHTTP(response, r)
-	assert.Equal(t, response.Code, http.StatusUnauthorized) //token is not provided/invalid
-	assert.Equal(t, response.Body.String(), "{\"message\":\"Request didn't have a valid authentication token.\"}\n")
+	assert.Equal(t, http.StatusUnauthorized, response.Code) //token is not provided/invalid
+	assert.Equal(t, "{\"message\":\"Request didn't have a valid authentication token.\"}\n", response.Body.String())
```
Comment (jlpadilla) [inline pkg/rbac/authenticationMiddleware_test.go:54]: Updated order of args for more accurate test failure logs.
```
assert.Equal(t, expected, actual)
```

**Review by Anxhela21 — APPROVED** [review]

**Pull request merged**